File: Leetcode/0242 Valid Anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        int char_count[26] = {};
        
        for(char c : s) char_count[c - 'a']++;
        
        for(char c : t) char_count[c - 'a']--;
        
        for(int i = 0;i < 26;++i)
            if(char_count[i])
                return false;
        
        return true;
    }
};

File: Leetcode/0242 Valid Anagram.ts
function isAnagram(s: string, t: string): boolean {
    const sCharCount = getCharCount(s), tCharCount = getCharCount(t);
    return sCharCount.every((c, i) => c === tCharCount[i]);
}

function getCharCount(s: string): number[] {
    const charCount = new Array<number>(26).fill(0);
    for (const c of s) {
        const idx = c.charCodeAt(0) - 'a'.charCodeAt(0);
        charCount[idx]++;
    }
    return charCount;
}
